stablehlo/reference/ProcessGrid.cpp

```diff
@@ -60,6 +76,8 @@ ProcessGroups ProcessGrid::crossReplica(
     SmallVector<SmallVector<uint32_t>> replicaGroups) {
   ProcessGroups processGroups;
   for (auto replicaGroup : replicaGroups) {
+    if (replicaGroup.empty())
+      llvm::report_fatal_error("empty replicaGroup found during crossReplica");
```

Comment:
Why is this a problem? This is not covered in the spec. (Same below for other methods implementing communication strategies).

Reply:
The original constraint is
* (C3) `0 <= replica_groups < size(replica_groups)`.

This implies the updated constraint:
* (C3 new) `1 <= size(replica_groups)`.
and
* (C2 old) `size(replica_groups)` is defined as:
  * `num_replicas` if `cross_replica` is used.
  * `num_replicas` if `cross_replica_and_partition` is used.
  * `num_processes` if `flattened_ids` is used.

And as per https://github.com/openxla/stablehlo/issues/498#issuecomment-1331396793, replica_groups cannot be empty. If this is checked in the verifier the checks here aren't needed. I guess this circles back to #498. I'll leave the spec alone for now and let the ticket address this as a whole.